pkg/ddc/base/operation_lock.go

```diff
@@ -0,0 +1,135 @@
+/*
+Copyright 2022 The Fluid Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package base
+
+import (
+	"context"
+	"fmt"
+	"github.com/fluid-cloudnative/fluid/pkg/common"
+	"github.com/fluid-cloudnative/fluid/pkg/dataoperation"
+	cruntime "github.com/fluid-cloudnative/fluid/pkg/runtime"
+	"github.com/fluid-cloudnative/fluid/pkg/utils"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/util/retry"
+	"reflect"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func GetDataOperationRef(name, namespace string) string {
+	// namespace may contain '-', use '/' as separator
+	return fmt.Sprintf("%s/%s", namespace, name)
+}
+
+// LockTargetDataset lock target dataset if not locked by this data operation.
+func LockTargetDataset(ctx cruntime.ReconcileRequestContext, object client.Object,
+	operation dataoperation.OperationInterface, engine Engine) error {
+	targetDataset := ctx.Dataset
+
+	operationTypeName := string(operation.GetOperationType())
+
+	// 1. Check if there's any conflict
+	conflictDataOpRef := targetDataset.GetLockedNameForOperation(operationTypeName)
+	dataOpRef := GetDataOperationRef(object.GetName(), object.GetNamespace())
```

Comment:
`types.NamespacedName` can replace the `GetDataOperationRef` function. `types.NamespacedName` is a struct that has Namespace and Name fields, which can be used to represent the name and namespace of an object. 

For example,

```golang
        conflictingDataOpKey := targetDataset.GetLockedNameForOperation(operationTypeName)
	dataOpRef := GetDataOperationRef(object.GetName(), object.GetNamespace())

	dataOpKey := types.NamespacedName{
		Namespace: object.GetNamespace(),
		Name:      object.GetName(),
	}

	// If the target dataset is already locked by this operation, return nil
	if dataOpKey.String()  == conflictingDataOpKey{
		return nil
	}
```

Reply:
rename the `GetDataOperationRef` to `GetDataOperationKey` and inside use the types.NamespacedName to generate key.